module/Decision/src/Decision/Controller/FileBrowser/LocalFileReader.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+namespace Decision\Controller\FileBrowser;
+
+/**
+ * Description of LocalFileReader
+ *
+ * @author s134399
+ */
+class LocalFileReader implements FileReader
+{
+
+    /**
+     * The location in the local filesystem that is considered the 'root' for this browser
+     * @var string
+     */
+    static private $root;
+
+    public function __construct($root)
+    {
+        $this->root = $root;
+    }
+
+    public function downloadFile($path)
+    {
+        $fullPath = $this->root . $path;
+        if (!is_file($fullPath)) {
+            return null;
+        }
+        $response = new \Zend\Http\Response\Stream();
+        $response->setStream(fopen('file://' . $fullPath, 'r'));
+        $response->setStatusCode(200);
+        $headers = new \Zend\Http\Headers();
+        $headers->addHeaderLine('Content-Type', 'application/octet-stream')
+                ->addHeaderLine('Content-Disposition', 'attachment; filename="' . end(explode('/', $fullPath)) . '"')
```

Comment:
If you leave out `attachment`, then the browser will just display the file rather than downloading it immediately.

Reply:
Ah, I just copied this from somewhere. I should do that.

Reply:
Btw, attachment isn't enough (on Firefox, at least). Dropping 'application' as well seems to do the trick, though.